src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -0,0 +1,713 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.Decorator;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Utilities;
+using AutoRest.CSharp.Output.Builders;
+using AutoRest.CSharp.Output.Models.Serialization.Json;
+using System.Text.Json;
+using AutoRest.CSharp.Mgmt.Generation;
+using AutoRest.CSharp.Mgmt.Models;
+using AutoRest.CSharp.Output.Models.Requests;
+using System.Text.Json.Serialization;
+using Azure.Core.Serialization;
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    internal partial class MgmtBaseTestWriter: MgmtClientBaseWriter
+    {
+        public static CodeWriter _tagsWriter = new CodeWriter();
+        public static HashSet<string>  variableNames = new HashSet<string>();
+
+        public MgmtBaseTestWriter(CodeWriter writer, MgmtTypeProvider provider, BuildContext<MgmtOutputLibrary> context) : base(writer, provider, context)
+        {
+        }
+
+        public void WriteTestDecorator()
```

Comment:
It is called `Attribute` in DotNet
```suggestion
        public void WriteTestAttributes()
```